src/anyio/_core/_subprocesses.py

```diff
@@ -75,3 +92,142 @@ async def open_process(command: Union[str, Sequence[str]], *, stdin: int = PIPE,
+    ``terminateProcess()`` on Windows).
+
+    :param func: a callable
+    :param args: positional arguments for the callable
+    :param cancellable: ``True`` to allow cancellation of the operation while it's running
+    :param limiter: capacity limiter to use to limit the total amount of processes running
+        (if omitted, the default limiter is used)
+    :return: an awaitable that yields the return value of the function.
+
+    """
+    async def send_raw_command(pickled_cmd: bytes):
+        try:
+            await stdin.send(pickled_cmd)
+            response = await buffered.receive_until(b'\n', 50)
+            status, length = response.split(b' ')
+            if status not in (b'RETURN', b'EXCEPTION'):
+                raise RuntimeError(f'Worker process returned unexpected response: {response!r}')
+
+            pickled_response = await buffered.receive_exactly(int(length))
+        except BaseException as exc:
+            workers.discard(process)
+            try:
+                process.kill()
+                await process.wait()
+            except ProcessLookupError:
+                pass
+
+            raise BrokenWorkerProcess from exc
+
+        retval = pickle.loads(pickled_response)
+        if status == b'EXCEPTION':
+            assert isinstance(retval, BaseException)
+            raise retval
+        else:
+            return retval
+
+    # First pickle the request before trying to reserve a worker process
+    await checkpoint_if_cancelled()
+    request = pickle.dumps(('run', func, args), protocol=pickle.HIGHEST_PROTOCOL)
+
+    # If this is the first run in this event loop thread, set up the necessary variables
+    try:
+        workers = _process_pool_workers.get()
+        idle_workers = _process_pool_idle_workers.get()
+    except LookupError:
+        workers = set()
+        idle_workers = deque()
+        _process_pool_workers.set(workers)
+        _process_pool_idle_workers.set(idle_workers)
+        get_asynclib().setup_process_pool_exit_at_shutdown(workers)
+
+    async with (limiter or current_default_worker_process_limiter()):
+        # Pop processes from the pool (starting from the most recently used) until we find one that
+        # hasn't exited yet
+        process: Process
+        while idle_workers:
+            process, idle_since = idle_workers.pop()
+            if process.returncode is None:
+                stdin = cast(ByteSendStream, process.stdin)
+                buffered = BufferedByteReceiveStream(cast(ByteReceiveStream, process.stdout))
+
+                # Prune any other workers that have been idle for WORKER_MAX_IDLE_TIME seconds or
+                # longer
+                now = current_time()
+                killed_processes: List[Process] = []
+                while idle_workers:
+                    if now - idle_workers[0][1] < WORKER_MAX_IDLE_TIME:
+                        break
+
+                    process, idle_since = idle_workers.pop()
```

Comment:
Should this be `popleft`?

Reply:
The idea is to pop the most recently added idle worker. Popleft() would pop the least recent one.

Reply:
For lifo caching that's right, of course, but this appears to be the pruning step. If you do not start from the oldest idle worker, you will kill all the idle workers as soon as the oldest one expires!

Reply:
Oops, fixing!